portal-kernel/test/unit/com/liferay/portal/kernel/scheduler/messaging/SchedulerEventMessageListenerWrapperTest.java

```diff
@@ -185,19 +240,19 @@ public void testConcurrentReceiveWithTimeout() throws Exception {
 
 		@Override
 		public void receive(Message message) {
-			if (_lock.tryLock()) {
-				try {
-					_waitCountDownLatch.countDown();
-
-					_blockCountDownLatch.await();
-
-					message.setResponse(message);
-				}
-				catch (InterruptedException ie) {
-				}
-				finally {
-					_lock.unlock();
-				}
+			_lock.lock();
```

Comment:
Hi @yuhai,
Why you made this change?

The reason using tryLock here is to make sure that listener is not invoked concurrently, if it is invoked concurrently, then only one message is process.

Tina.

Reply:
Hi @tinatian 

When exposed the bug in the first commit, I wanted to express the receive() executed and then Assert.assertNull(_testMessage2.getResponse()) won't pass. In addition, I thought SchedulerEventMessageListenerWrapper had controlled the listener is not invoked concurrently. So did the change was for test using.

Thanks,
Hai